prometheus_juju_exporter/collector.py

```diff
@@ -137,14 +129,19 @@ async def _get_machine_stats(
         :param str model_name: the name of the model the machines are in
         :param str gauge_name: the name of the gauge
         """
+        virt_mac_prefixes = self.config["machine"]["virt_macs"].get(str).split(",")
+
         for machine in machines.values():
             # check machine type with the prefix of its mac address
             interfaces = machine["network-interfaces"].values()
             machine_type = MachineType.METAL
 
             for i in interfaces:
                 mac_addresses = i["mac-address"]
-                if mac_addresses.startswith(tuple(self.prefixes)):
+                if (
+                    mac_addresses.startswith(tuple(virt_mac_prefixes))
+                    and virt_mac_prefixes[0] != ""
```

Comment:
if this is intended to be validation of config values, I'd suggest moving it close to where you read the config (line 132)

Reply:
This is now being handled by confuse library's [StrSeq](https://github.com/beetbox/confuse/blob/46504692f9f6c26e248a90243062ccca1fd29992/confuse/templates.py#L354) function as virt_macs value is a list now